client/data/services/QuestionDataService.js

```diff
@@ -64,6 +64,49 @@ tieData.factory('QuestionDataService', [
             questionId);
         }
         return QuestionObjectFactory.create(globalData.questions[questionId]);
+      },
+      /**
+       * Returns the question's user friendly title.
+       *
+       * @param {string} questionId
+       * @returns {string}
+       */
+      getQuestionTitle(questionId) {
+        var question = this.getQuestion(questionId);
+        return question.getTitle();
+      },
+      /**
+       * Returns the question's shortened instructions, which includes
+       * only the text content in the first task (ignores code blocks).
+       *
+       * @param {string} questionId
+       * @returns {string} concatenated string of all the text segments
+       */
+      getQuestionPreviewInstructions(questionId) {
+        var question = this.getQuestion(questionId);
+        var tasks = question.getTasks();
```

Comment:
Perhaps `var firstTask = question.getTasks()[0];`, so that you don't have to do [0] all the time.

Or even: `var instructionsForFirstTask = question.getTasks()[0].getInstructions()`.